test/jdk/jdk/jfr/event/io/TestSocketChannelEvents.java

```diff
@@ -102,6 +103,7 @@ public void xrun() throws IOException {
                 readerThread.start();
 
                 try (SocketChannel sc = SocketChannel.open(ssc.getLocalAddress())) {
+                    addExpectedEvent(IOEvent.createSocketConnectEvent(sc.socket()));
```

Comment:
This is SocketChannel in blocking mode where the connect succeeds. There is also the non-blocking and where connect fails. In addition the connection can established with the socket adaptor. So 6 possible cases for SocketChannel if the test is expanded.

Reply:
yes, testing still to be done